webpack/ForemanTasks/Components/TasksTable/TasksTable.js

```diff
@@ -0,0 +1,116 @@
+import React, { useEffect } from 'react';
+import { Spinner } from 'patternfly-react';
+import PropTypes from 'prop-types';
+import { Table } from 'foremanReact/components/common/table';
+import { STATUS } from 'foremanReact/constants';
+import MessageBox from 'foremanReact/components/common/MessageBox';
+import { translate as __ } from 'foremanReact/common/I18n';
+import Pagination from 'foremanReact/components/Pagination/PaginationWrapper';
+import createTasksTableSchema from './TasksTableSchema';
+import { updateURlQuery } from './TasksTableHelpers';
+import './TasksTable.scss';
+
+const TasksTable = ({
+  getTableItems,
+  error,
+  status,
+  results,
+  query,
+  itemCount,
+  pagination,
+  setSort,
+  setPagination,
+  sort,
+}) => {
+  useEffect(() => {
+    getTableItems();
+  }, [
+    query.state,
+    query.result,
+    query.mode,
+    query.time,
+    pagination.page,
+    pagination.perPage,
+    sort.by,
+    sort.order,
+  ]);
+  const onPageChange = ({ page, perPage }) => {
+    updateURlQuery({
+      page,
+      per_page: perPage,
+    });
+    setPagination({ page, perPage });
```

Comment:
should we use `async & await` here?

Reply:
why?

Reply:
I thought you were updating the url and only after that the table will get updated with pagination,
same as with the query, but now I see that it is only the action that updates it in this case, thanks